### ISSUE TITLE
The --cloudconfig CLI option  has an innaccurate name

### ISSUE THREAD

**Issue by tbarsballe** [planetlabs/planet-client-python#970]
The `--cloudconfig` option of the `planet orders create` CLI command has an inaccurate name, as it is not limited to cloud delivery configurations. 

It is entirely possible to provide a local delivery config using the `--cloudconfig` option, e.g.:
```
{  
    "archive_type": "zip",
    "single_archive": true,
    "archive_filename": "{{name}}_{{order_id}}.zip"
}
```

The option should be named `--delivery` or `--deliveryconfig`, and [the docs](https://planet-sdk-for-python-v2.readthedocs.io/en/latest/cli/cli-orders/#cloud-delivery) updated accordingly. The [equivalent SDK method](https://planet-sdk-for-python-v2.readthedocs.io/en/latest/python/sdk-reference/#planet.order_request.delivery) is more accurately named `delivery`

**Comment by tbarsballe** [comment]
With the actual bug report out of the way - I fully understand this is is an exceedingly nitpicky bug report, and frankly user experience may be better if this options remains as-is, so don't feel bad about closing this as Wontfix.

**Comment by sgillies** [comment]
@tbarsballe I agree that `--delivery` would be easier to understand and explain. `--cloudconfig` comes from v1 of the CLI when the cloud config bits and archive format were separate options.

#963 is also related to delivery. Maybe we'll fix both of these for 2.1.0.

I imagine resolving this one by adding a `--delivery` option and leaving `--cloudconfig`, but with a deprecation warning and an admonition to switch to `--delivery`.
